modules/api/pkg/handler/v2/external_cluster/kubeone.go

```diff
@@ -291,3 +298,36 @@ func getKubeOneAPIMachineDeployments(ctx context.Context,
 
 	return nodeDeployments, nil
 }
+
+func getKubeOneClusterDetails(ctx context.Context,
+	apiCluster *apiv2.ExternalCluster,
+	masterClient ctrlruntimeclient.Client,
+	cluster *kubermaticv1.ExternalCluster,
+	clusterProvider provider.ExternalClusterProvider) error {
+	var containerRuntime, region string
+
+	clusterClient, err := clusterProvider.GetClient(ctx, masterClient, cluster)
+	if err != nil {
+		return err
+	}
+
+	nodes, err := kuberneteshelper.ListControlPlaneNode(ctx, clusterClient)
+	if err != nil {
+		return err
+	}
+
+	for _, node := range nodes.Items {
+		region = node.Labels[NodeRegionLabel]
+		containerRuntimeVersion := node.Status.NodeInfo.ContainerRuntimeVersion
+		strSlice := strings.Split(containerRuntimeVersion, ":")
+		containerRuntime = strSlice[0]
+
+		clusterSpec := &apiv2.KubeOneClusterSpec{
+			Region: region,
+		}
+		apiCluster.Spec.ContainerRuntime = containerRuntime
```

Comment:
Similar problem here. We have a loop that is filling the same variable attribute. It looks like it shouldn't be a loop.